### DISCUSSION TITLE
retention rules

### DISCUSSION THREAD

**Discussion by ewebsk** [jsuto/piler#193] [General]
guessing how retention rules works, I entered in admin gui retention rule for spam messages to 1 day but see that they are still stored in metadata with default_retention_days expiration in retained field, what is right way when I will purge spam messages sooner as other mails?

**Comment by jsuto** [comment]
Run pilertest on the given message to see what piler thinks about it. Also be sure to fill only the relevant fields in the retention rule. What's not relevant, leave that empty. Eg. if the message subject doesn't matter, then don't put an asterisk or anything to the subject field.

**Reply by ewebsk** [reply]
in metadata table is message marked as spam but retained field is set to default_retention_days, after creating retention rule is not needed restart piler or so?

**Comment by jsuto** [comment]
If you change the rules, you need to restart piler daemon.

**Comment by ewebsk** [comment]
rebooted whole vps and now it works as expected, guessing how work archiving rules is possible with archiving rules remove spam messages before they enter archive?

**Comment by jsuto** [comment]
Yes, you can exclude recognized spam from the archive. That's why recent piler relesese renamed the "archiving rules" to "exclusion rules".
